src/main.py

```diff
@@ -43,51 +43,59 @@ def get_block_info(block_name):
     return jsonify(block.get_info())
 
 
-@app.route('/<block_name>', methods=['POST'], strict_slashes=False)
-@app.route('/<block_name>/resume', methods=['POST'], strict_slashes=False)
-def execute_block(block_name):
-    body = request.get_json()
-    is_resuming = request.path.endswith(('/resume', '/resume/'))
-
-    if body is None:
-        return 'Body must be a JSON object', HTTPStatus.BAD_REQUEST
-
-    try:
-        block = block_manager.blocks[block_name]
-    except KeyError:
-        return 'No block with that name exists.', HTTPStatus.NOT_FOUND
+def get_requested_params(block, request_body, is_resuming):
+    if is_resuming and 'requested_params' in request_body['state']:
+        requested_params = request_body['state']['requested_params']
+    elif not is_resuming:
+        requested_params = block.get_info()['params']
+    else:
+        requested_params = dict()
+    return requested_params
 
-    if is_resuming and 'state' not in body:
-        return 'State must be provided when resuming', HTTPStatus.BAD_REQUEST
 
+def clean_params(block, request_body, is_resuming):
     # Create new dictionary that will only contain parameters included in block specification
     cleaned_params = dict()
 
-    if is_resuming and 'requested_params' in body['state']:
-        requested_params = body['state']['requested_params']
-    elif not is_resuming:
-        requested_params = block.get_info()['params']
-    else:
-        requested_params = dict()
+    requested_params = get_requested_params(block, request_body, is_resuming)
 
-    if 'params' not in body and len(requested_params):
+    if 'params' not in request_body and len(requested_params):
         return 'No parameters sent', HTTPStatus.BAD_REQUEST
```

Comment:
clean_params cant return Http responses. Maybe rather raise an exception here and catch it in execute_block.